lib/temporal.rb

```diff
@@ -81,6 +84,54 @@ def signal_workflow(workflow, signal, workflow_id, run_id, input = nil)
       )
     end
 
+    # Long polls for a workflow to be completed and returns whatever the execute function
+    # returned.
+    # run_id of nil: await the entire workflow completion.  This can span multiple runs
+    # in the case where the workflow uses continue as new.
+    def await_workflow_result(workflow, workflow_id:, run_id: nil, **args)
+      options = args.delete(:options) || {}
+      execution_options = ExecutionOptions.new(workflow, options)
+
+      current_run_id = run_id
+      loop do
+        history_response = client.get_workflow_execution_history(
+          namespace: execution_options.namespace,
+          workflow_id: workflow_id,
+          run_id: current_run_id,
+          wait_for_new_event: true,
+          event_type: :close
+        )
+        history = Workflow::History.new(history_response.history.events)
+        closed_event = history.events.first
+        case closed_event.type
+        when 'WORKFLOW_EXECUTION_COMPLETED'
+          payloads = closed_event.attributes.result
+          return nil if !payloads # happens when the workflow itself returns nil
```

Comment:
`from_result_payloads` should be able to handle this use case just fine, no need to special case it